ephios/extra/auth.py

```diff
@@ -1,14 +1,36 @@
+from typing import Any, Dict
+from urllib.parse import urljoin
+
+import jwt
+from django.conf import settings
 from django.contrib.auth import get_user_model
-from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from django.contrib.auth.backends import ModelBackend
+from django.core.exceptions import SuspiciousOperation
+from django.urls import reverse
+from jwt import InvalidTokenError
+from oauthlib.oauth2 import WebApplicationClient
+from requests_oauthlib import OAuth2Session
+from urllib3.exceptions import RequestError
+
+from ephios.core.models.users import IdentityProvider
 
 
-class EphiosOIDCAB(OIDCAuthenticationBackend):
+class EphiosOIDCAB(ModelBackend):
+    def decode_jwt_token(self, token: str) -> Dict[str, Any]:
+        jwks_client = jwt.PyJWKClient(self.provider.jwks_uri)
+        header = jwt.get_unverified_header(token)
+        key = jwks_client.get_signing_key(header["kid"]).key
+        decoded = jwt.decode(token, key, [header["alg"]], audience=self.provider.client_id)
+        return decoded
+
     def create_user(self, claims):
         user = get_user_model()(email=claims.get("email"))
         user.set_unusable_password()
         user.first_name = claims.get("given_name", "")
         user.last_name = claims.get("family_name", "")
```

Comment:
I don't like how this creates an account without any name when using e.g. nextcloud. Have you thought about what to do about this?

Reply:
the only viable option would be using the full name and doing some magic splitting there which is not ideal as well

Reply:
we could ask the user to provide name/date of birth when logging in for the first time?!

Reply:
sounds good. But I would keep that separate from this PR as the problem is already there and not changed by this PR. Also, we could use that to think about some kind of user onboarding flow in general

Reply:
to be solved with #146